src/tallies/tally_scoring.cpp

```diff
@@ -0,0 +1,2347 @@
+              score -= bank.E;
+            } else if (bank.particle == Particle::Type::positron) {
+              // Annihilation of the positron will produce two new photons
+              score -= 2*MASS_ELECTRON_EV;
+            }
+          }
+          score *= p->wgt_last_ * flux;
+        } else {
+          // Calculate photon heating cross section on-the-fly
+          if (i_nuclide >= 0) {
+            // Find the element corresponding to the nuclide
+            auto name = data::nuclides[i_nuclide]->name_;
+            std::string element = to_element(name);
+            int i_element = data::element_map[element];
+            auto& heating {data::elements[i_element].heating_};
+            auto i_grid = p->photon_xs_[i_element].index_grid;
+            auto f = p->photon_xs_[i_element].interp_factor;
+            score = std::exp(heating(i_grid) + f * (heating(i_grid+1) -
+              heating(i_grid))) * atom_density * flux;
+          } else {
+            if (p->material_ != MATERIAL_VOID) {
+              const Material& material {*model::materials[p->material_]};
+              for (auto i = 0; i < material.nuclide_.size(); ++i) {
+                auto i_element = material.element_[i];
+                auto atom_density = material.atom_density_(i);
+                auto& heating {data::elements[i_element].heating_};
+                auto i_grid = p->photon_xs_[i_element].index_grid;
+                auto f = p->photon_xs_[i_element].interp_factor;
+                score += std::exp(heating(i_grid) + f * (heating(i_grid+1) -
+                  heating(i_grid))) * atom_density * flux;
+              }
+            }
+          }
+        }
+      }
+      break;
+
+    default:
+      if (tally.estimator_ == ESTIMATOR_ANALOG) {
+        // Any other score is assumed to be a MT number. Thus, we just need
+        // to check if it matches the MT number of the event
+        if (p->event_mt_ != score_bin) continue;
+        score = p->wgt_last_*flux;
+      } else {
+        // Any other cross section has to be calculated on-the-fly
+        if (score_bin < 2) fatal_error("Invalid score type on tally "
+          + std::to_string(tally.id_));
+        score = 0.;
+        if (i_nuclide >= 0) {
+          const auto& nuc {*data::nuclides[i_nuclide]};
+          auto m = nuc.reaction_index_[score_bin];
+          if (m == C_NONE) continue;
+          const auto& rxn {*nuc.reactions_[m]};
+          auto i_temp = p->neutron_xs_[i_nuclide].index_temp;
+          if (i_temp >= 0) { // Can be false due to multipole
+            auto i_grid = p->neutron_xs_[i_nuclide].index_grid;
+            auto f = p->neutron_xs_[i_nuclide].interp_factor;
+            const auto& xs {rxn.xs_[i_temp]};
+            if (i_grid >= xs.threshold) {
+              score = ((1.0 - f) * xs.value[i_grid-xs.threshold]
+                + f * xs.value[i_grid-xs.threshold+1]) * atom_density * flux;
+            }
+          }
+        } else {
```

Comment:
Optional: This `else` block can be combined with the `if` block below, to slightly simplify the code.

```
} else if (p->material_ != MATERIAL_VOID) {
  ...
}
```